databricks/koalas/indexing.py

```diff
@@ -625,12 +638,16 @@ def __setitem__(self, key, value):
                     kdf[temp_key_col] = rows_sel
                 if isinstance(value, Series):
                     kdf[temp_value_col] = value
-                kdf = kdf.sort_values(temp_natural_order)
+                kdf = kdf.sort_values(temp_natural_order).drop(temp_natural_order)
 
                 if isinstance(rows_sel, Series):
-                    rows_sel = kdf[temp_key_col]
+                    rows_sel = F.col(
+                        "`{}`".format(kdf[temp_key_col]._internal.data_spark_column_names[0])
```

Comment:
I wonder if we should just document that we don't support backquotes in the column names ..